Clients/GUI/Pages/InstrumentGraphic.razor

```diff
@@ -0,0 +1,71 @@
+﻿@page "/instrument/{Figi}"
+@using System.Timers
+@using DTO.BrokerRequests
+@using Microsoft.AspNetCore.SignalR.Client
+@inject NavigationManager NavigationManager
+@inject ISessionStorageService Storage
+@inject IJSRuntime JSRuntime
+
+    <div class="container ml-0 align-items-center w-75" id="myChart">@errorMessage</div>
+
+@code {
+
+    [Parameter]
+    public string Figi { get; set; }
+
+    public HubConnection hubConnection;
+    private string errorMessage = "";
+    private Timer timer;
+    private string token = "";
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            token = await Storage.GetItemAsync<string>("tinkoffToken");
+            if (token == null)
+            {
+                errorMessage = "Access error";
+                StateHasChanged();
+                return;
+            }
+
+            hubConnection = new HubConnectionBuilder()
```

Comment:
Not sure, but it wouldn't be better to connect to hub in the other class? In razor page it is not the best way to see logic

Reply:
graphic changed. Hub moved in the other class